### PR TITLE
Make workerinfo stored as json

### PR DESCRIPTION
**Pull request by lucyge2022**
### What changes are proposed in this pull request?

Store workerinfo as json format on etcd for EtcdMembershipManager.

### Why are the changes needed?

For cross-language worker membership retrieval.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

File: dora/core/common/src/main/java/alluxio/membership/DefaultServiceEntity.java [hunk 1]
```diff
@@ -75,23 +78,55 @@ public CloseableClient getKeepAliveClient() {
...
-   * @param dis
-   * @throws IOException
+   * Convert a DefaultServiceEntity into a json string.
+   * @param entity
+   * @return json string
    */
-  public void deserialize(DataInputStream dis) throws IOException {
-    mServiceEntityName = dis.readUTF();
-    mRevision = dis.readLong();
+  public static String toJson(DefaultServiceEntity entity) {
```
Comment (Kai-Zhang) [inline dora/core/common/src/main/java/alluxio/membership/DefaultServiceEntity.java:125]: maybe leave the implement to decide the serialization type? license uses the service entity and it's no need to use json for now. maybe we can provide a interface like this?
```java
public static byte[] serialize(DefaultServiceEntity entity) throws IOException {...}
public static DefaultServiceEntity deserialize(byte[] buf) throws IOException {...}
```

File: dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java [hunk 2]
```diff
@@ -60,57 +61,69 @@ public class ServiceDiscoveryRecipe {
...
+  private final ConcurrentHashMap<String, DefaultServiceEntity> mRegisteredServices
       = new ConcurrentHashMap<>();
 
   /**
    * CTOR for ServiceDiscoveryRecipe.
    * @param client
    * @param clusterIdentifier
    */
-  public ServiceDiscoveryRecipe(AlluxioEtcdClient client, String clusterIdentifier) {
+  ServiceDiscoveryRecipe(AlluxioEtcdClient client, String clusterIdentifier) {
```
Comment (Kai-Zhang) [inline dora/core/common/src/main/java/alluxio/membership/ServiceDiscoveryRecipe.java:75]: license engine may need to construct its own `ServiceDiscoveryRecipe`. is there any context on this change?

**Review by LuQQiu — APPROVED** [review]
LGTM, thanks for supporting json worker info!

**Comment by lucyge2022** [comment]
alluxio-bot, merge this please.

**Comment by dbw9580** [comment]
any particular consideration for choosing Gson over Jackson?

**Comment by lucyge2022** [comment]
> any particular consideration for choosing Gson over Jackson?

actually no, i saw Gson was used in codebase and has much simpler syntax.

**Pull request merged**